src/toniarts/openkeeper/world/creature/CreatureControl.java

```diff
@@ -538,22 +549,42 @@ public short getOwnerId() {
     public void navigateToAssignedTask() {
 
         Vector2f loc = assignedTask.getTarget(this);
-        if (loc != null) {
-            GraphPath<TileData> outPath = worldState.findPath(worldState.getTileCoordinates(getSpatial().getWorldTranslation()), new Point((int) Math.floor(loc.x), (int) Math.floor(loc.y)), creature);
+        if (loc == null) {
+            return;
+        }
 
-            if (outPath != null && outPath.getCount() > 1) {
+        GraphPath<TileData> outPath = worldState.findPath(getSpatial().getWorldTranslation(), new Point((int) Math.floor(loc.x), (int) Math.floor(loc.y)), creature);
+        PrioritySteering<Vector2> prioritySteering = new PrioritySteering(this, 0.001f);
 
-                // Debug
-//                worldHandler.drawPath(new LinePath<>(pathToArray(outPath)));
-                PrioritySteering<Vector2> prioritySteering = new PrioritySteering(this, 0.0001f);
-                FollowPath<Vector2, LinePathParam> followPath = new FollowPath(this, new LinePath<>(pathToArray(outPath), true), 2);
-                followPath.setDecelerationRadius(1f);
-                followPath.setArrivalTolerance(0.2f);
-                prioritySteering.add(followPath);
+        if (outPath != null && outPath.getCount() > 1) {
 
-                prioritySteering.setEnabled(!isAnimationPlaying());
-                setSteeringBehavior(prioritySteering);
-            }
+            // Debug
+//            if (Main.isDebug()) {
+//                worldState.getApp().enqueue(() -> {
+//                    worldState.drawPath(waypoints);
+//                });
+//            }
+
+            FollowPath<Vector2, LinePathParam> followPath = new FollowPath(this, new LinePath<>(getWaypoints(outPath), true), 2);
+            followPath.setDecelerationRadius(0.3f);

```

Comment:
imps steel cut corners when moves. https://gamedevelopment.tutsplus.com/tutorials/understanding-steering-behaviors-path-following--gamedev-8769 Smoothing the Movement header. Where this attribute in gdx-ai ?